src/types/Post.ts

```diff
@@ -0,0 +1,11 @@
+import { User } from './User';
+import { Comment } from './Comment';
+
+export interface Post {
+  userId: number;
+  id: number;
+  title: string;
+  body: string;
+  user: User | null;
+  userComments: Comment[] | null;
```

Comment:
Метод `filter` не повертає null, а повертає пустий масив
```suggestion
  userComments: Comment[];
```